crates/testing/src/helpers.rs

```diff
@@ -67,18 +67,46 @@ pub async fn build_system_handle<
     let builder: TestDescription<TYPES, I, V> = TestDescription::default_multiple_rounds();
 
     let launcher = builder.gen_launcher(node_id);
+    build_system_handle_from_launcher(node_id, &launcher).await
+}
 
+/// create the [`SystemContextHandle`] from a node id and `TestLauncher`
+/// # Panics
+/// if cannot create a [`HotShotInitializer`]
+pub async fn build_system_handle_from_launcher<
+    TYPES: NodeType<InstanceState = TestInstanceState>,
+    I: NodeImplementation<
+            TYPES,
+            Storage = TestStorage<TYPES>,
+            AuctionResultsProvider = TestAuctionResultsProvider<TYPES>,
+        > + TestableNodeImplementation<TYPES>,
+    V: Versions,
+>(
+    node_id: u64,
+    launcher: &TestLauncher<TYPES, I, V>,
+) -> (
+    SystemContextHandle<TYPES, I, V>,
+    Sender<Arc<HotShotEvent<TYPES>>>,
+    Receiver<Arc<HotShotEvent<TYPES>>>,
+) {
     let network = (launcher.resource_generator.channel_generator)(node_id).await;
     let storage = (launcher.resource_generator.storage)(node_id);
     let marketplace_config = (launcher.resource_generator.marketplace_config)(node_id);
-    let config = launcher.resource_generator.config.clone();
+    let mut config = launcher.resource_generator.config.clone();
 
     let initializer = HotShotInitializer::<TYPES>::from_genesis::<V>(TestInstanceState::new(
-        launcher.metadata.async_delay_config,
+        launcher.metadata.async_delay_config.clone(),
     ))
     .await
     .unwrap();
 
+    // See whether or not we should be DA
+    let is_da = node_id < config.da_staked_committee_size as u64;
```

Comment:
is it possible to take this from `config.known_da_nodes` instead?

I feel like it would be better to have `config.known_da_nodes` initialized to the first `da_staked_committee_size` nodes, and take directly from that here (otherwise we have da nodes getting set independently in two different places, which might lead to surprising behaviour down the line)

Reply:
Yeah we should do this. However the code is like this all over the test suites, I'll create an issue for this one and follow up since it might be a bigger change.